### ISSUE TITLE
Validation batch labels wrong

### ISSUE THREAD

**Issue by tonystark12** [ultralytics/yolov5#6344]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### YOLOv5 Component

Validation

### Bug

I noticed that the labels mentioned in the runs/train/exp/val_batch_labels images are wrong, while the predictions seem to right. This will provide wrong feedback to the training and thus work on a loss that is false. 

I am training on a dataset to detect car dents and the accuracy seems to be perfectly fine in val predictions, but wrong truth labels are put up. To cross check with the dataset, I did verify that the labels mentioned in the .txt files are correct, but something's amiss while loading them probably. 

Please help me in figuring this out. Attached are the batch images of valid with preds, that result in /runs/train/exp
The images are in the order : Val_labels / Val_Preds

![val_batch0_labels](https://user-images.githubusercontent.com/4528392/150060115-fc5cb4d5-0fbe-4bb8-8c3c-e9513064d623.jpg)
![val_batch0_pred](https://user-images.githubusercontent.com/4528392/150060132-eda2ca83-35ec-4ce0-b46e-e51309ca8ef5.jpg)
![val_batch1_labels](https://user-images.githubusercontent.com/4528392/150060155-d36f53b0-ba41-469c-ab5b-36a7366a90e1.jpg)
![val_batch1_pred](https://user-images.githubusercontent.com/4528392/150060167-f3723118-0403-4065-8cf2-688289f63b74.jpg)
![val_batch2_labels](https://user-images.githubusercontent.com/4528392/150060175-6193b707-eff6-44f1-b396-63b5c3d03e8f.jpg)
![val_batch2_pred](https://user-images.githubusercontent.com/4528392/150060177-cc9de070-6a95-4a67-aa6d-d6c1c973d344.jpg)

### Environment

I am working with
python 3.8.3
Win 11
torch 1.7
tf 2.4.0
cuda 11

### Minimal Reproducible Example

It is regular training. The Valid set was in COCO format. I uploaded the COCO format into roboflow and then obtained the yolov5 format. I can provide the valid set with labels for cross check or the link to roboflow if required.

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by yeldarby** [comment]
Hey @tonystark12 - I'm one of the developers of Roboflow & am in this area of the codebase today anyway. If you could [share your dataset with Roboflow support](https://docs.roboflow.com/support/sharing-a-workspace-with-roboflow-support) I'd be happy to check and see if this is a bug on our end or not.

**Comment by lana211191** [comment]
hello, did you end up finding the problem? i am having a similar issue

**Comment by TommiA** [comment]
I have also exactly the same issue. The validation batch image collage carriers wrong labels. I am not really sure if this affects the precision in the end but it seems systematic. E.g. Jaguars are labelled as Austins in the vehicle domain?!

![val_batch0_labels](https://user-images.githubusercontent.com/18534149/182906145-68dd6e9e-e1f0-49f7-928d-bd2245a105fc.jpg)

**Comment by glenn-jocher** [comment]
@TommiA this is pretty simple. If the visuals show incorrect labels then your dataset labels are incorrect. That's the purpose of the visual.

To train correctly your data must be in YOLOv5 format. Please see our [Train Custom Data](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading) tutorial for full documentation on dataset setup and all steps required to start training your first model. A few excerpts from the tutorial:

### 1.1 Create dataset.yaml

[COCO128](https://www.kaggle.com/datasets/ultralytics/coco128) is an example small tutorial dataset composed of the first 128 images in [COCO](http://cocodataset.org/#home) train2017. These same 128 images are used for both training and validation to verify our training pipeline is capable of overfitting. [data/coco128.yaml](https://github.com/ultralytics/yolov5/blob/master/data/coco128.yaml), shown below, is the dataset config file that defines 1) the dataset root directory `path` and relative paths to `train` / `val` / `test` image directories (or *.txt files with image paths), 2) the number of classes `nc` and 3) a list of class `names`:
```yaml
# Train/val/test sets as 1) dir: path/to/imgs, 2) file: path/to/imgs.txt, or 3) list: [path/to/imgs1, path/to/imgs2, ..]
path: ../datasets/coco128  # dataset root dir
train: images/train2017  # train images (relative to 'path') 128 images
val: images/train2017  # val images (relative to 'path') 128 images
test:  # test images (optional)

# Classes
nc: 80  # number of classes
names: [ 'person', 'bicycle', 'car', 'motorcycle', 'airplane', 'bus', 'train', 'truck', 'boat', 'traffic light',
         'fire hydrant', 'stop sign', 'parking meter', 'bench', 'bird', 'cat', 'dog', 'horse', 'sheep', 'cow',
         'elephant', 'bear', 'zebra', 'giraffe', 'backpack', 'umbrella', 'handbag', 'tie', 'suitcase', 'frisbee',
         'skis', 'snowboard', 'sports ball', 'kite', 'baseball bat', 'baseball glove', 'skateboard', 'surfboard',
         'tennis racket', 'bottle', 'wine glass', 'cup', 'fork', 'knife', 'spoon', 'bowl', 'banana', 'apple',
         'sandwich', 'orange', 'broccoli', 'carrot', 'hot dog', 'pizza', 'donut', 'cake', 'chair', 'couch',
         'potted plant', 'bed', 'dining table', 'toilet', 'tv', 'laptop', 'mouse', 'remote', 'keyboard', 'cell phone',
         'microwave', 'oven', 'toaster', 'sink', 'refrigerator', 'book', 'clock', 'vase', 'scissors', 'teddy bear',
         'hair drier', 'toothbrush' ]  # class names
```


### 1.2 Create Labels

After using a tool like [Roboflow Annotate](https://roboflow.com/annotate?ref=ultralytics) to label your images, export your labels to **YOLO format**, with one `*.txt` file per image (if no objects in image, no `*.txt` file is required). The `*.txt` file specifications are:

- One row per object
- Each row is `class x_center y_center width height` format.
- Box coordinates must be in **normalized xywh** format (from 0 - 1). If your boxes are in pixels, divide `x_center` and `width` by image width, and `y_center` and `height` by image height.
- Class numbers are zero-indexed (start from 0).

<img width="800" alt="Image Labels" src="https://user-images.githubusercontent.com/26833433/91506361-c7965000-e886-11ea-8291-c72b98c25eec.jpg">

The label file corresponding to the above image contains 2 persons (class `0`) and a tie (class `27`):

<p align="center"><img width="428" src="https://user-images.githubusercontent.com/26833433/112467037-d2568c00-8d66-11eb-8796-55402ac0d62f.png"></p>


### 1.3 Organize Directories

Organize your train and val images and labels according to the example below. YOLOv5 assumes  `/coco128` is inside a `/datasets` directory **next to** the `/yolov5` directory. **YOLOv5 locates labels automatically for each image** by replacing the last instance of `/images/` in each image path with `/labels/`. For example:
```bash
../datasets/coco128/images/im0.jpg  # image
../datasets/coco128/labels/im0.txt  # label
```

<p align="center"><img width="700" src="https://user-images.githubusercontent.com/26833433/134436012-65111ad1-9541-4853-81a6-f19a3468b75f.png"></p>
</details>


Good luck 🍀 and let us know if you have any other questions!
